### ISSUE TITLE
Export with pricacy rules generates empty and unlinked objects

### ISSUE THREAD

**Issue by Jefferson49** [fisharebest/webtrees#3817]
When using privacy rules during GEDCOM export, the generated export contains several empty and unlinked objects.

The attached example contains an example tree in GEDCOM file "tree.txt".
When exporting this tree from webtrees 2.0.15 with pricacy rules, the attached file "tree exported as visitor with privacy applied.txt" is generated.

The exported example file contains the following empty or unlinked objects:
0 @X21@ INDI
1 SEX F
0 @F3@ FAM
0 @S1@ SOUR
0 @SUB1@ SUBM
0 @X22@ OBJE
0 @X23@ OBJE
0 @X24@ OBJE

This is only a short example. If I export a larger database, I get several hundreds of empty objects.

The (almost empty) and unlinked "@X21@ INDI" record should not be exported at all, because @X21@ is restricted by privacy.

While empty FAM, SOUR might be allowed but ugly, the generated GEDCOM for OBJE an SUBM is not compliant to the GEDCOM 5.5.1 standard, because a minimum sub-structure is required, e.g.
0 @SUB1@ SUBM
1 NAME xyz
0 @X22@ OBJE
1 FILE Foto.jpg
2 FORM jpg

Of course, it would not make sense to add any kind of sub-structure. The prefered way would be to suppress the export of empty objects.

[tree.txt](https://github.com/fisharebest/webtrees/files/6246023/tree.txt)
[tree exported as visitor with privacy applied.txt](https://github.com/fisharebest/webtrees/files/6246024/tree.exported.as.visitor.with.privacy.applied.txt)
![Privacy settings in webtrees v2](https://user-images.githubusercontent.com/81484983/113344061-e4b96280-9330-11eb-821c-c6b9d052843a.jpg)

**Comment by fisharebest** [comment]
This logic has existed for a long time.  IIRC, it was done for performance reasons - the privacy calculations can use a lot of resources and this allows larger files to be processed on smaller servers.

It may be worth reviewing this.

**Comment by fisharebest** [comment]
Closing this issue in favour of #4883

**Comment by Jefferson49** [comment]
After the discussion in https://github.com/fisharebest/webtrees/issues/4883, I gained a lot more insights and better understanding about the background of empty GEDCOM objects in the webtrees export. Based on this information, I do not see a solution and agree to close the issue.

On the other hand, I still consider the empty objects in my GEDCOM export as "ugly". However, I was able to solve this by findig a very valuable tool suite for post-processing of my GEDCOM exports: [Gedcom Service Programs](https://ofb.hesmer.name/gedserpro_e.html)
